var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -73,15 +73,41 @@ def patch(self):
     @property
     def blas_libs(self):
         shared = True if '+shared' in self.spec else False
+        query_parameters = self.spec.last_query.extra_parameters
+        query2libraries = {
+            tuple(): ['blas'],
+            ('c',): [
+                'libcblas',
+                'libblas',
```

Comment:
i guess it's matter of convention, but i would return here `libcblas` only. Assuming that user is supposed to add fortran part if needed. In your case in `plasma` you were also specifying `c` interface separately.

Reply:
@alalazo @adamjstewart @tgamblin opinions? 

p.s. for `hdf5` `hl` is only high level libs, without `fortran` or `c`. So i think it make sense to return here `c` part only.

Reply:
I think its best to request the highest level in the depencency chain - and let the package [aka netlib-lapack] tell what the full list of libraries are [instead of plasma or similar packages trying to construct that list from individual pieces]

```
$ ldd ./libcblas.so 
	linux-vdso.so.1 (0x00007ffd5d372000)
	libblas.so.3 => /home/balay/tmp/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/netlib-lapack-3.6.1-x2ltjbn36sqbyuwjfvzmwn5ko2vin7ew/lib64/libblas.so.3 (0x00007f5069ad2000)
	libgfortran.so.4 => /lib64/libgfortran.so.4 (0x00007f50696f6000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f50693a1000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f506918a000)
	libquadmath.so.0 => /lib64/libquadmath.so.0 (0x00007f5068f4a000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f5068b65000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f5069f41000)
$ ldd liblapacke.so 
	linux-vdso.so.1 (0x00007ffe6371d000)
	liblapack.so.3 => /home/balay/tmp/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/netlib-lapack-3.6.1-x2ltjbn36sqbyuwjfvzmwn5ko2vin7ew/lib64/liblapack.so.3 (0x00007f60cbcb0000)
	libblas.so.3 => /home/balay/tmp/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/netlib-lapack-3.6.1-x2ltjbn36sqbyuwjfvzmwn5ko2vin7ew/lib64/libblas.so.3 (0x00007f60cba5b000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f60cb676000)
	libgfortran.so.4 => /lib64/libgfortran.so.4 (0x00007f60cb29a000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f60caf45000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f60cad2e000)
	libquadmath.so.0 => /lib64/libquadmath.so.0 (0x00007f60caaee000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f60cc8d6000)
```

So I prefer the current code - but if you want me to change it - I can do that.

Reply:
but you can also request `lapack:c,fortran` and it's all good? I think returning `c` only is simply more descriptive and more flexible in case some fancy build system requires `c` and `fortran` parts to be specified separately?

Reply:
let's see what others have to say...

Reply:
Sure `lapack:c,fortran` will also be fine